### ISSUE TITLE
"Notice: Undefined offset: 0 " from VarnishPurger when purging a single tag

### ISSUE THREAD

**Issue by varacti** [api-platform/core#3893]
**API Platform version(s) affected**: 2.5.8

**Description**
An exception is thrown from `ApiPlatform\Core\HttpCache\VarnishPurger` line 70 with the message "Notice: Undefined offset: 0" when purging a single tag.

The purged tags are defined in class `ApiPlatform\Core\Bridge\Doctrine\EventListener\PurgeHttpCacheListener` as an associative array [ string => string ], for example in function `gatherResourceAndItemTags` :
```
114     $iri = $this->iriConverter->getIriFromResourceClass($resourceClass);
        // $iri is supposed to be a string here because of getIriFromResourceClass signature !!!
115     $this->tags[$iri] = $iri;
```
Then this array is directly send to VarnishPurger from PurgeHttpCacheListener :
```
106     $this->purger->purge($this->tags);
```

But if this array contains a single asociation, VarnishPurger fails line 70 and throws an exception.

**Possible Solution**  
Replace "ApiPlatform\Core\HttpCache\VarnishPurger" Line 70 :
```
yield $parts[0];
```
with:
```
yield array_shift($parts)
```

**Comment by julienfalque** [comment]
Woops, likely related to #3843, I'll have a look.

**Comment by julienfalque** [comment]
See #3894.

**Comment by julienfalque** [comment]
This issue can be closed as #3894 was merged.
